File: cpp/024.cpp
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode *dummy = new ListNode(0);
        ListNode *tail = dummy;
        while (head != nullptr && head->next != nullptr) {
            ListNode *temp = head->next->next;
            head->next->next = nullptr;

            tail->next = head->next;
            tail->next->next = head;
            head->next = nullptr;

            tail = tail->next->next;

            head = temp;
        }
        if (head != nullptr)
            tail->next = head;
        return dummy->next;
    }
};

File: go/024.go
func swapPairs(head *ListNode) *ListNode {
    dummy := &ListNode{}
    tail := dummy
    for head != nil && head.Next != nil {
        temp := head.Next.Next
        head.Next.Next = nil

        tail.Next = head.Next
        tail.Next.Next = head
        head.Next = nil

        tail = tail.Next.Next

        head = temp
    }
    if head != nil {
        tail.Next = head
    }
    return dummy.Next
}
